accounts/templates/accounts/profile.html

```diff
@@ -28,96 +24,205 @@ <h2>{{request.user}}</h2>
       </div>
     </div>
 
     <!-- Middle Column of Lists -->
     <div class="middle_col">
+      <!-- Watch later list-->
       <div class="list_div">
-        <h3>Watchlist</h3>
+        <h3>Watch Later</h3>
         <div class="list" id="watchlist">
-          <p class="placeholder_text">No movies yet!</p>
+          {% if watchlist_movies %}
+            {% for movie in watchlist_movies %}
+              <img src="https://image.tmdb.org/t/p/original{{ movie.poster }}" onclick="goToMoviePage('{{ movie.pk }}')" alt="Movie Poster" />
+            {% endfor %}
+          {% else %}
+            <p class="placeholder_text">No movies yet!</p>
+          {% endif %}
         </div>
       </div>
+      <!-- Movies watched list-->
       <div class="list_div">
         <h3>Watched</h3>
-        <div class="list" id="liked_movies">
-          <p class="placeholder_text">No movies yet!</p>
+        <div class="list" id="watched_movies">
+          {% if watched_movies %}
+            {% for movie in watched_movies %}
+              <img src="https://image.tmdb.org/t/p/original{{ movie.poster }}" onclick="goToMoviePage('{{ movie.pk }}')" alt="Movie Poster" />
+            {% endfor %}
+          {% else %}
+            <p class="placeholder_text">No movies yet!</p>
+          {% endif %}
         </div>
       </div>
     </div>
 
-    <!-- Friends List Container & Right Column -->
+    <!-- Friends List Container / Right Column -->
     <div class="friends_div">
       <div class="friends_title">
         <h2>Your Friends</h2>
-        <button class="btn btn-open">Add Friend</button>
+        <button class="btn btn-open" onclick="openAddFriendModal()">Add Friend</button>
       </div>
+      <!--Friends List-->
       <div class="friends_list">
-        <!-- Placeholder Friends -->
-        <div class="friend">
-          <img
-            src="{% static 'accounts/images/profile.png' %}"
-            class="profile_img"
-          />
-          <h3>Friend Name</h3>
-        </div>
-        <div class="friend">
-          <img
-            src="{% static 'accounts/images/profile.png' %}"
-            class="profile_img"
-          />
-          <h3>Friend Name</h3>
-        </div>
-        <div class="friend">
-          <img
-            src="{% static 'accounts/images/profile.png' %}"
-            class="profile_img"
-          />
-          <h3>Friend Name</h3>
-        </div>
+        {% if friends %}
+          {% for friend in friends %}
+            
+            <div class="friend">
+              <img
+                src='{{ friend.profile_picture.url }}'
+                class="profile_img" onclick="openRemoveFriendModal('{{ friend.username }}')"
+              />
+              <h3>{{ friend.username }}</h3>
+            </div>
+          {% endfor %}
+        {% else %}
+          <p class="placeholder_text">No friends yet!</p>
+        {% endif %}
+      </div>
+      <!--Recieved Friend Requests-->
+      <h2>Recieved Requests</h2>
```

Comment:
could we not just have one big ol' like "pending friend requests" that would be like:
'incoming' 'outgoing' underneath them or something?

Reply:
Hmm is this to save on space? I think it's more cleaner/convenient to have the requests separated by sections rather than in 1 combined section since the user knows that a specific section belongs to a specific type of request (given by the header h2 tag) instead of looking at the text under each profile (which I think can also cause a lot of text clutter).